src/framework/matrix.hpp

```diff
@@ -513,6 +566,16 @@ inline void matrix<T>::SetOutputStyle(enum OutputStyle outputstyle) {
   // sets the outputstyle
   outputstyle_ = outputstyle;
 }
+
+template<class T>
+bool matrix<T>::AlmostEqual(const matrix<T>& rhs) const {
+  for(auto i = 0; i < rhs.size(); ++i){
+    if(complex_almost_equal<T>(mat_[i], rhs[i]) == false)
```

Comment:
Does this also work if matrices are real? 

Also could this be moved to a util function using templates: `almost_equal(matrix<T>&, matrix<T>&)` so we can have one `almost_equal` function that works with most types.

Reply:
Yep! Better having overloaded functions.